### PR TITLE
Add OS pid and processID to the logs

### PR DESCRIPTION
**Pull request by espadolini**
With this PR we add the `component` field early to the logs, we make sure that `process.log` is the same as `process.Config.Log` (used in quite a few places), and we add a new `pid` field (visible when logging with the json formatter) that contains `<OS pid from os.Getpid>.<incrementing process ID as generated by nextProcessID()>`, which is very useful to understand what's going on when there's multiple `TeleportProcess` instances around - this happens during restarts/upgrades and during tests (especially integration tests).

Product questions:
- Do we want to log the OS pid ourselves? Is that considered "sensitive"?
- Should we always add the field at every level, or only at DEBUG? It might be something that most users won't care about, but it's not a lot of extra data. OTOH, we'd need to document why the `pid` has an extra `.1`, `.2`, etc.
- Are name and format ok? Should we have two numerical fields instead?

edit: product questions answered as per https://github.com/gravitational/teleport/pull/13190#pullrequestreview-998874217 okaying the current implementation

### CODE REVIEW

**Review by GavinFrazar — APPROVED** [review]
LGTM but I have no answers for your product questions :)

File: lib/service/service.go [hunk 1]
```diff
@@ -4194,6 +4196,10 @@ func validateConfig(cfg *Config) error {
 		cfg.Console = io.Discard
 	}
 
+	if cfg.Log == nil {
+		return trace.BadParameter("config: please supply logger")
```
Comment (zmb3) [inline lib/service/service.go:4200]: Why can't we just create a logger if one was not provided?

**Review by zmb3 — APPROVED** [review]
No concern with the approach - discussed this in our product sync and everyone 👍.

See my comment about the logger though.

File: lib/service/service.go (hunk 1 above)
Comment (espadolini) [reply lib/service/service.go:4200]: What should the default logger be in that situation? Currently we just panic at the first use of `cfg.Log` or `process.log`.

Comment (zmb3) [reply lib/service/service.go:4200]: We do this in several places:

```
	if cfg.Log == nil {
		logger = log.StandardLogger()
	}
```

**Review by timothyb89 — APPROVED** [review]

**Pull request merged**